DSCResources/MSFT_HybridRunbookWorker/MSFT_HybridRunbookWorker.psm1

```diff
@@ -311,7 +311,21 @@ function TestRegModule
 #>
 function TestRegRegistry
 {
-    Test-Path -Path HKLM:\SOFTWARE\Microsoft\HybridRunbookWorker
+    $Reg = Get-ItemProperty -Path HKLM:\SOFTWARE\Microsoft\HybridRunbookWorker
+    if ($null -eq $Reg) {
+        #newer version
+        Get-Item -Path HKLM:\SOFTWARE\Microsoft\HybridRunbookWorker\*\* | 
+            ForEach-Object -Process {
+                if (Get-ChildItem -Path $_.PSPath |
+                    Where-Object -FilterScript {
+                        $_.PSChildName -eq 'User'
```

Comment:
The output type of this helper function is mend to be a ```bool```. Could you change the logic a bit so that there are code paths that actually return a bool?